### PR TITLE
[MISC,FEATURE] Allow char* searches in string indices

### PR DESCRIPTION
**Pull request by eseiler**
* Removes deprecated includes
* Switches to range version of `size`, `begin`, `end`
* Adds static_assert to check for compatibility of index and query alphabet (60 lines of trace with a meaningful message in the beginning vs. 600 lines with no clear message)
* Adds overloads for `char const * const` and `std::initializer<char const * const>` that allow searches on string indices for, e.g., `"query"` or `{"query1", "query2"}`
* Adds tests for the overloads

### CODE REVIEW

File: include/seqan3/search/algorithm/search.hpp [hunk 1]
```diff
@@ -95,6 +98,25 @@ inline auto search(index_t const & index, queries_t && queries, configuration_t
...
+
+//! \overload
+template <FmIndex index_t, typename configuration_t>
+inline auto search(index_t const & index,
+                   std::initializer_list<char const * const> const & queries,
+                   configuration_t const & cfg)
+{
+    std::vector<std::basic_string_view<char>> query;
+    query.reserve(std::ranges::size(queries));
+    ranges::for_each(queries, [&query](char const * const q){query.push_back(std::basic_string_view{q});});
```
Comment (smehringer) [inline include/seqan3/search/algorithm/search.hpp]: std::ranges::for_each

File: include/seqan3/search/algorithm/search.hpp [hunk 2]
```diff
@@ -95,6 +98,25 @@ inline auto search(index_t const & index, queries_t && queries, configuration_t
...
+    return search(index, std::basic_string_view{queries}, cfg);
+}
+
+//! \overload
+template <FmIndex index_t, typename configuration_t>
+inline auto search(index_t const & index,
+                   std::initializer_list<char const * const> const & queries,
+                   configuration_t const & cfg)
+{
+    std::vector<std::basic_string_view<char>> query;
```
Comment (smehringer) [inline include/seqan3/search/algorithm/search.hpp]: `std::basic_string_view<char>` == `std::string_view`

File: include/seqan3/search/algorithm/search.hpp [hunk 3]
```diff
@@ -95,6 +98,25 @@ inline auto search(index_t const & index, queries_t && queries, configuration_t
...
+
+//! \overload
+template <FmIndex index_t, typename configuration_t>
+inline auto search(index_t const & index,
+                   std::initializer_list<char const * const> const & queries,
+                   configuration_t const & cfg)
+{
+    std::vector<std::string_view> query;
+    query.reserve(std::ranges::size(queries));
+    std::ranges::for_each(queries, [&query](char const * const q){query.push_back(std::string_view{q});});
```
Comment (h-2) [inline include/seqan3/search/algorithm/search.hpp]:
```suggestion
    std::ranges::for_each(queries, [&query] (char const * const q) { query.push_back(std::string_view{q}); });
```

File: include/seqan3/search/algorithm/search.hpp [hunk 4]
```diff
@@ -128,6 +153,23 @@ inline auto search(index_t const & index, queries_t && queries)
...
+    return search(index, std::string_view{queries});
+}
+
+//! \overload
+template <FmIndex index_t>
+inline auto search(index_t const & index, std::initializer_list<char const * const> const & queries)
+{
+    std::vector<std::string_view> query;
+    query.reserve(std::ranges::size(queries));
+    std::ranges::for_each(queries, [&query](char const * const q){query.push_back(std::string_view{q});});
```
Comment (h-2) [inline include/seqan3/search/algorithm/search.hpp]: spacing see above

**Comment by eseiler** [comment]
@h-2 Can you have a last look, please?

**Pull request merged**